src/user-guide/running-workflows/workflow-completion.rst

```diff
@@ -3,43 +3,37 @@
 Workflow Completion
 ===================
 
-A workflow is complete, and the scheduler will automatically
-:term:`shut down <shutdown>`, if no tasks remain in the
-:term:`n=0 <n-window>`.
-
-That is, all active tasks have finished, and no tasks remain waiting on
-:term:`prerequisites <prerequisite>` or "external" constraints (such as
-:term:`xtriggers <xtrigger>` or task :term:`hold`).
-
-If no active tasks remain and all external constraints are satisfied,
-but the n=0 window contains tasks waiting with partially satisfied
-:term:`prerequisites <prerequisite>`, or tasks with :term:`final status` and
-:term:`incomplete outputs <output completion>`, then the workflow is
-not complete and the scheduler will :term:`stall` pending manual intervention.
-
+A workflow can :term:`shut down <shutdown>` once all
+:term:`active tasks <active task>` complete without spawning further
+downstream activity - i.e., when :term:`n=0 window <n-window>` empties out.
 
 .. _scheduler stall:
 
 Scheduler Stall
 ===============
 
-A stalled workflow has not run to :term:`completion <workflow completion>`
-but cannot continue without manual intervention. 
+If there are no tasks waiting on as yet unsatisfied external constraints
+such clock and xtriggers, and all activity has ceased but workflow has
+not :ref:`run to completion <workflow completion>`, then it
+has stalled and requires manual intervention to continue.
```

Comment:
Hemmingway editor gives this a reading age of "Post-graduate". This suggestion is US grade 7 (13-14 YO).

```suggestion
A workflow has stalled if:

* No tasks are waiting on unstatisfied external events, like clock triggers and xtriggers.
* AND All activity has ceased.
* AND The workflow has not run to completion.

A workflow which has stalled requires manual intervention to continue.
```